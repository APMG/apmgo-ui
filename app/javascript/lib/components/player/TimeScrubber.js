// @flow
import React from 'react'
import { dispatch, connect } from 'react-redux'
import { timeScrubberChange, updatePlayTime } from '../../redux/audio-player'
import Slider from 'rc-slider'

type TimeScrubberProps = {
  paused: boolean,
  currentTime: number,
  duration: number,
  updatePlayTime: (playtime: number) => {},
  timeScrubberChange: (currentTime: number) => {}
}

export default class TimeScrubberPresenter extends React.Component {

  props: TimeScrubberProps
  state: {
    timeDraggingPoint: number|false
  }

  constructor(props: TimeScrubberProps) {
    super(props)
    this.state = {
      timeDraggingPoint: false
    }
  }

  timeScrubberValue() {
    if(!this.props.duration || isNaN(this.props.duration)) {
      return 0
    }

    let val = this.props.currentTime // / this.props.duration
    return val // <= 1 ? val : 1
  }

  // EVENT HANDLERS

  _timeChange(newValue: number) {
    if (this.state.timeDraggingPoint) {
      this.setState({
        timeDraggingPoint: newValue
      })
      this.props.updatePlayTime(newValue) //this.props.duration * )
    } else {
      this.setState({
        timeDraggingPoint: newValue
      })
    }
  }

  _timeChangeDragStop() {
    if (!this.state.timeDraggingPoint) {
      return
    }

    if (this.state.timeDraggingPoint) {
      let currentTime = this.state.timeDraggingPoint //* this.props.duration
      this.props.timeScrubberChange(currentTime)
    }

    this.setState({
      timeDraggingPoint: false
    })
  }

  render() {
    return (
      <div>
        <Slider
          max={this.props.duration}
          min={0}
          value={this.timeScrubberValue()}
          onChange={ this._timeChange.bind(this) }
          onAfterChange={this._timeChangeDragStop.bind(this) }
        />
      </div>
    )
  }
}
