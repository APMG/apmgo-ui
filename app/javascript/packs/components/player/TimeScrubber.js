// @flow
import React from 'react'
import { dispatch, connect } from 'react-redux'
import { timeScrubberChange, updatePlayTime } from '../../redux/audio-player'
import Slider from 'material-ui/Slider'

type TimeScrubberProps = {
  paused: boolean,
  currentTime: number,
  duration: number,
  updatePlayTime: () => {},
  timeScrubberChange: () => {}
}

export class TimeScrubberPresenter extends React.Component {

  props: TimeScrubberProps
  state: {
    timeDraggingPoint: number|false
  }

  constructor(props: TimeScrubberProps) {
    super(props)
    this.state = {
      timeDraggingPoint: false
    }
  }

  timeScrubberValue() {
    if(!this.props.duration || isNaN(this.props.duration)) {
      return 0
    }

    let val = this.props.currentTime / this.props.duration
    return val <= 1 ? val : 1
  }

  // EVENT HANDLERS

  _timeChange(event: Event, newValue: number) {
    if (this.state.timeDraggingPoint) {
      this.setState({
        timeDraggingPoint: newValue
      })
      this.props.updatePlayTime(this.props.duration * newValue)
    } else {
      this.setState({
        timeDraggingPoint: newValue
      })
    }
  }

  _timeChangeDragStop() {
    if (!this.state.timeDraggingPoint) {
      return
    }

    if (this.state.timeDraggingPoint) {
      let currentTime = this.props.duration * this.state.timeDraggingPoint
      this.props.timeScrubberChange(currentTime)
    }

    this.setState({
      timeDraggingPoint: false
    })
  }

  render() {
    return (
      <div>
        <Slider
          value={ this.timeScrubberValue() }
          onChange={ this._timeChange.bind(this) }
          onDragStop={this._timeChangeDragStop.bind(this) }
        />
      </div>
    )
  }
}

// const mapDispatchToProps = (dispatch, ownProps) => {
//   return {
//     updatePlayTime: (currentTime) => {
//       dispatch(updatePlayTime(ownProps.item_id, Math.ceil(currentTime)))
//     },
//     timeScrubberChange: (currentTime) => {
//       dispatch(timeScrubberChange(ownProps.item_id, Math.ceil(currentTime)))
//     }
//   }
// }

// const mapStateToProps = (state) => {
//   return {
//     currentTime: state.audioPlayer.currentTime,
//     duration: state.audioPlayer.duration,
//     paused: state.audioPlayer.paused
//   }
// }

export default TimeScrubberPresenter //connect(mapStateToProps, mapDispatchToProps)(TimeScrubberPresenter)
