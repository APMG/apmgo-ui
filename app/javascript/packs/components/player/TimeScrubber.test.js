
import React from 'react'
import getSnapshotJson from '../../redux/__tests__/mock-initial-state'
import TimeScrubber, { TimeScrubberPresenter } from './TimeScrubber'

describe('Time Scrubber Component', () => {
  describe('Presenter', () => {
    it('Renders', () => {
      let tree = getSnapshotJson(<TimeScrubberPresenter/>, {audioPlayer: { paused: true }})
      expect(tree).toMatchSnapshot()
    })
  })
})
