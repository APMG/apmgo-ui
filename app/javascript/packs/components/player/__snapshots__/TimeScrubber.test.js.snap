// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Time Scrubber Component Presenter Renders 1`] = `
<div>
  0:00
  <div
    style={
      Object {
        "muiPrepared": true,
      }
    }
    value={0}
  >
    <span />
    <span />
    <div
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "cursor": "default",
          "height": 18,
          "marginBottom": 48,
          "marginTop": 24,
          "muiPrepared": true,
          "position": "relative",
          "touchCallout": "none",
          "userSelect": "none",
          "width": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "height": 2,
            "left": 0,
            "muiPrepared": true,
            "position": "absolute",
            "top": 8,
            "width": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "[object Object]": 0,
              "backgroundColor": "#00bcd4",
              "height": "100%",
              "left": 0,
              "marginRight": 6,
              "muiPrepared": true,
              "position": "absolute",
              "transition": "margin 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "width": "calc(0%)",
            }
          }
        />
        <div
          style={
            Object {
              "[object Object]": 0,
              "backgroundColor": "#bdbdbd",
              "height": "100%",
              "marginLeft": 6,
              "muiPrepared": true,
              "position": "absolute",
              "right": 0,
              "transition": "margin 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "width": "calc(100%)",
            }
          }
        />
        <div
          style={
            Object {
              "backgroundClip": "padding-box",
              "backgroundColor": "#ffffff",
              "border": "2px solid #bdbdbd",
              "borderRadius": "50%",
              "boxShadow": "none",
              "boxSizing": "border-box",
              "cursor": "pointer",
              "height": 12,
              "left": "0%",
              "margin": "1px 0 0 0",
              "muiPrepared": true,
              "outline": "none",
              "overflow": "visible",
              "pointerEvents": "inherit",
              "position": "absolute",
              "top": 0,
              "transform": "translate(-50%, -50%)",
              "transition": "background 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms, border-color 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms, width 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms, height 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "width": 12,
              "zIndex": 1,
            }
          }
          tabIndex={0}
        >
          <div
            style={
              Object {
                "height": 12,
                "left": -2,
                "muiPrepared": true,
                "overflow": "visible",
                "position": "absolute",
                "top": -2,
                "width": 12,
              }
            }
          />
        </div>
      </div>
    </div>
    <input
      max={1}
      min={0}
      name={undefined}
      required={true}
      step={0.01}
      type="hidden"
      value={0}
    />
  </div>
</div>
`;
